`LoadError: AssertionError: Don't understand this multi-output result of the graph` when loading model

### Package Version

0.2.0

### Julia Version

1.7.1

### OS / Environment

MacOs

### Describe the bug

I'm trying to load this model https://github.com/chembl/chembl_multitask_model and getting `LoadError: AssertionError: Don't understand this multi-output result of the graph`



### Steps to Reproduce

```julia
using ONNX

path = "chembl_31_multitask.onnx"

# dummy input
dummy = rand(Float32, 1024, 1)

# load the model
mt_chembl = ONNX.load(path, dummy)
```

the model is available here: https://github.com/chembl/chembl_multitask_model/tree/main/trained_models/chembl_31_model

### Expected Results

model to load and hopefully outputting some predictions!

### Observed Results

Error

### Relevant log output

```
ERROR: LoadError: AssertionError: Don't understand this multi-output result of the graph
Stacktrace:
 [1] load(io::IOStream, args::Matrix{Float32}; backends::Vector{Symbol}, exec::Bool)
   @ ONNX ~/.julia/packages/ONNX/VJnjv/src/load.jl:285
 [2] #138
   @ ~/.julia/packages/ONNX/VJnjv/src/load.jl:294 [inlined]
 [3] open(f::ONNX.var"#138#139"{Vector{Symbol}, Bool, Tuple{Matrix{Float32}}}, args::String; kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
   @ Base ./io.jl:330
 [4] open
   @ ./io.jl:328 [inlined]
 [5] #load#137
   @ ~/.julia/packages/ONNX/VJnjv/src/load.jl:293 [inlined]
 [6] load(filename::String, args::Matrix{Float32})
   @ ONNX ~/.julia/packages/ONNX/VJnjv/src/load.jl:293
 [7] top-level scope
   @ ~/projects/onnx_julia/onnx_julia.jl:18
```

--- a/Project.toml
+++ b/Project.toml
@@ -1,6 +1,6 @@
 name = "ONNX"
 uuid = "d0dd6a25-fac6-55c0-abf7-829e0c774d20"
-version = "0.2.0"
+version = "0.2.1"
 
 [deps]
 EnumX = "4e289a0a-7415-4d19-859d-a7e5c4648b56"

--- a/src/load.jl
+++ b/src/load.jl
@@ -311,12 +311,19 @@ function load(io::IO, args...; backends=[:ONNX], exec::Bool=true)
     if length(g.output) == 1
         tape.result = Umlaut.bound(tape, V(length(tape)))
     else
-        # tuple output: we expect tape to contain these outputs as vars  destructured
-        # from a multi-ouput op using a sequence of `getfield()` calls
         vars = [tape.c.name2var[o.name] for o in g.output]
-        @assert(all(tape[v] isa Call && tape[v].fn == getfield for v in vars),
-            "Don't understand this multi-output result of the graph")
-        tape.result = tape[vars[1]].args[1]
+        is_unpacked_tuple = all(
+            tape[v] isa Call && tape[v].fn == getfield && tape[v].args[1] === tuple
+            for v in vars
+        )
+        if is_unpacked_tuple
+            # tuple output: we expect tape to contain these outputs as vars  destructured
+            # from a multi-ouput op using a sequence of `getfield()` calls
+            tape.result = tape[vars[1]].args[1]
+        else
+            # independent vars in the ouput - create a new tuple var
+            tape.result = push!(tape, mkcall(tuple, vars...))
+        end
     end
     return tape
 end

--- a/src/save.jl
+++ b/src/save.jl
@@ -281,6 +281,17 @@ function save_node!(g::GraphProto, ::@opconfig_kw(:ONNX, onnx_concat), op::Umlau
     push!(g.node, nd)
 end
 
+function save_node!(g::GraphProto, ::OpConfig{:ONNX, typeof(tuple)}, op::Umlaut.Call)
+    @assert(
+        op.id == op.tape.result.id,
+        "tuple() doesn't have a corresponding ONNX op and is only allowed as " *
+        "the result of the tape, in which case it represents multiple outputs " *
+        "of the graph"
+    )
+    # do nothing
+end
+
+
 ##############################################################################
 #                                    API                                     #
 ##############################################################################
@@ -315,7 +326,7 @@ function save(io::IO, tape::Tape{ONNXCtx})
     if res.val isa Tuple
         # if the last operation in the graph is multi-output, there must be
         # unpacked elements of that var
-        vars = unpacked_vars(res)
+        vars = res.fn === tuple ? res.args : unpacked_vars(res)
         @assert(all(v isa V for v in vars), "Cannot save the tape because the result " *
             "is multi-output, but its elements aren't destructured to the tape")
         for v in vars

--- a/test/saveload.jl
+++ b/test/saveload.jl
@@ -3,6 +3,17 @@ import ONNX: NodeProto, ValueInfoProto, AttributeProto, onnx_name
 
 
 @testset "Save and Load" begin
+    @testset "Multioutput" begin
+        args = rand(3, 4), rand(3, 4)
+        tape = Tape(ONNXCtx())
+        inp = [push!(tape, Input(arg)) for arg in args]
+        out1 = push_call!(tape, ONNX.add, inp...)
+        out2 = push_call!(tape, ONNX.mul, inp...)
+        res = push_call!(tape, tuple, out1, out2)
+        tape.result = res
+        ort_test(tape, args...)
+    end
+
     @testset "Basic ops" begin
         args = (rand(3, 4), rand(3, 4))
         ort_test(ONNX.add, args...)

